lib/operations/list_collections.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+const CommandOperationV2 = require('./command_v2');
+const Aspect = require('./operation').Aspect;
+const defineAspects = require('./operation').defineAspects;
+const maxWireVersion = require('../core/utils').maxWireVersion;
+const listCollectionsTransforms = require('./db_ops').listCollectionsTransforms;
+const CONSTANTS = require('../constants');
+
+const LIST_COLLECTIONS_WIRE_VERSION = 3;
+
+class ListCollectionsOperation extends CommandOperationV2 {
+  constructor(db, filter, options) {
+    super(db, options);
+    this.options.full = true;
+
+    this.db = db;
+    this.filter = filter;
+    this.nameOnly = !!this.options.nameOnly;
+
+    if (typeof this.options.batchSize === 'number') {
+      this.batchSize = this.options.batchSize;
+    }
+  }
+
+  execute(server, callback) {
+    if (maxWireVersion(server) < LIST_COLLECTIONS_WIRE_VERSION) {
+      let filter = this.filter;
+      const databaseName = this.db.s.namespace.db;
+
+      // If we have legacy mode and have not provided a full db name filter it
+      if (
+        typeof filter.name === 'string' &&
+        !new RegExp('^' + databaseName + '\\.').test(filter.name)
+      ) {
+        filter = Object.assign({}, filter);
+        filter.name = this.db.s.namespace.withCollection(filter.name).toString();
+      }
+
+      // No filter, filter by current database
+      if (filter == null) {
+        filter.name = `/${databaseName}/`;
+      }
+
+      // Rewrite the filter to use $and to filter out indexes
+      if (filter.name) {
+        filter = { $and: [{ name: filter.name }, { name: /^((?!\$).)*$/ }] };
+      } else {
+        filter = { name: /^((?!\$).)*$/ };
+      }
+
+      const transforms = listCollectionsTransforms(databaseName);
+      server.query(
+        `${databaseName}.${CONSTANTS.SYSTEM_NAMESPACE_COLLECTION}`,
+        { query: filter },
+        { batchSize: this.batchSize || 1000 },
```

Comment:
Where is this default coming from?

Reply:
This is the default batch size for find operations. This whole section of code will go away once I can reuse a `FindOperation`, so I wouldn't worry _too_ much about it yet.